In [1]:
import pandas as pd
import numpy as np

# Json data cleaning: Scouting Reports 2019 - 2023

In [ ]:
# Read in the scouting report data for each year
recruits_2019 = pd.read_json(r'../Data/scouting_reports_2019.json')
recruits_2020 = pd.read_json(r'../Data/scouting_reports_2020.json')
recruits_2021 = pd.read_json(r'../Data/scouting_reports_2021.json')
recruits_2022 = pd.read_json(r'../Data/scouting_reports_2022.json')
recruits_2023 = pd.read_json(r'../Data/scouting_reports_2023.json')

### 2019 Data Cleaning
---

In [ ]:
# Take a peek at the 2019 data
recruits_2019.head()


,name,url,rank,position,height,weight,composite_rating,school_info,city,state,...,skill_delivery,skill_pocket_presence,skill_accuracy,skill_intangibles,skill_outside_the_pocket,skill_mobility,skill_arm_strength,skill_size,skill_playmaking_ability,skill_speed
0,Spencer Rattler,https://247sports.com/player/spencer-rattler-8...,1,PRO,6-1,198,99,"Pinnacle (Phoenix, AZ)",Phoenix,AZ,...,10.0,9.0,9.0,8.0,8.0,8.0,8.0,6.0,NaN,NaN
1,Jayden Daniels,https://247sports.com/player/jayden-daniels-89...,2,DUAL,6-3,175,97,"Cajon (San Bernardino, CA)",San Bernardino,CA,...,8.0,9.0,9.0,9.0,NaN,NaN,9.0,7.0,10.0,9.0
2,Bo Nix,https://247sports.com/player/bo-nix-77945/,3,DUAL,6-1.5,207,96,"Pinson Valley (Pinson, AL)",Pinson,AL,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Sam Howell,https://247sports.com/player/sam-howell-91707/,4,DUAL,6-0,225,95,"Sun Valley (Monroe, NC)",Monroe,NC,...,8.0,9.0,8.0,8.0,9.0,8.0,8.0,6.0,NaN,NaN
4,Max Duggan,https://247sports.com/player/max-duggan-87561/,5,DUAL,6-2,190,94,"Lewis Central (Council Bluffs, IA)",Council Bluffs,IA,...,9.0,9.0,9.0,10.0,NaN,NaN,8.0,8.0,9.0,10.0


Note: Specific issue with some players committed school being list as their high school instead of the college due to some of html code. Issue may be more prevlent for lower ranking players because they may not have commited anywhere out of high school, likely walk ons. Take a look at this below.

In [9]:
recruits_2019['committed_school'].value_counts()

committed_school
Arizona State Sun Devils    3
Alabama Crimson Tide        3
Ole Miss Rebels             3
Iowa State Cyclones         3
SMU Mustangs                3
                           ..
Northeast                   1
Norco                       1
Archbishop Moeller          1
Illinois State              1
Austin Peay                 1
Name: count, Length: 203, dtype: int64

In [16]:
df_2019 = recruits_2019[['name', 'committed_school']]

In [19]:
SCHOOL_MAP = {
    # FBS / major programs
    'Oklahoma Sooners':             'Oklahoma',
    'Arizona State Sun Devils':     'Arizona State',
    'Auburn Tigers':                'Auburn',
    'North Carolina Tar Heels':     'North Carolina',
    'TCU Horned Frogs':             'TCU',
    'South Carolina Gamecocks':     'South Carolina',
    'Wisconsin Badgers':            'Wisconsin',
    'Alabama Crimson Tide':         'Alabama',
    'Washington Huskies':           'Washington',
    'Oregon Ducks':                 'Oregon',
    'Arkansas Razorbacks':          'Arkansas',
    'Boise State Broncos':          'Boise State',
    'Michigan Wolverines':          'Michigan',
    'Texas Longhorns':              'Texas',
    'Ole Miss Rebels':              'Ole Miss',
    'California Golden Bears':      'California',
    'Arizona Wildcats':             'Arizona',
    'Florida Gators':               'Florida',
    'Tennessee Volunteers':         'Tennessee',
    'Oklahoma State Cowboys':       'Oklahoma State',
    'Minnesota Golden Gophers':     'Minnesota',
    'Cincinnati Bearcats':          'Cincinnati',
    'LSU Tigers':                   'LSU',
    'Notre Dame Fighting Irish':    'Notre Dame',
    'Washington State Cougars':     'Washington State',
    'Mississippi State':            'Mississippi State',
    'Toledo Rockets':               'Toledo',
    'Penn State Nittany':           'Penn State',
    'Texas A&M Aggies':             'Texas A&M',
    'Maryland Terrapins':           'Maryland',
    'Iowa State Cyclones':          'Iowa State',
    'Michigan State Spartans':      'Michigan State',
    'Pittsburgh Panthers':          'Pittsburgh',
    'Clemson Tigers':               'Clemson',
    'Southern Miss Golden':         'Southern Miss',
    'Southern Miss Golden Eagles':  'Southern Miss',
    'UCF Knights':                  'UCF',
    'Baylor Bears':                 'Baylor',
    'SMU Mustangs':                 'SMU',
    'Purdue Boilermakers':          'Purdue',
    'Missouri Tigers':              'Missouri',
    'Miami (OH) RedHawks':          'Miami (OH)',
    'Wake Forest Demon Deacons':    'Wake Forest',
    'BYU Cougars':                  'BYU',
    'Brigham Young Cougars':        'BYU',
    'USF Bulls':                    'USF',
    'Temple Owls':                  'Temple',
    'Miami Hurricanes':             'Miami',
    'Kent State Golden Flashes':    'Kent State',
    'FIU Golden Panthers':          'FIU',
    'Kentucky Wildcats':            'Kentucky',
    "Louisiana Ragin' Cajuns":      'Louisiana',
    "Louisiana Ragin'":             'Louisiana',
    'Northern Iowa Panthers':       'Northern Iowa',
    'Northern Iowa':                'Northern Iowa',
    'Georgia Tech Yellow':          'Georgia Tech',
    'Utah State Aggies':            'Utah State',
    'Nevada Wolf Pack':             'Nevada',
    'Richmond':                     'Richmond',
    'Kansas State Wildcats':        'Kansas State',
    'Texas Tech Red Raiders':       'Texas Tech',
    'Appalachian State':            'Appalachian State',
    'Boston College Eagles':        'Boston College',
    'Georgia State Panthers':       'Georgia State',
    'Memphis Tigers':               'Memphis',
    'Buffalo Bulls':                'Buffalo',
    'Syracuse Orange':              'Syracuse',
    'Old Dominion Monarchs':        'Old Dominion',
    'East Carolina Pirates':        'East Carolina',
    'Virginia Cavaliers':           'Virginia',
    'New Mexico State Aggies':      'New Mexico State',
    'Virginia Tech Hokies':         'Virginia Tech',
    'UCLA Bruins':                  'UCLA',
    'Hawaii Warriors':              'Hawaii',
    'Massachusetts Minutemen':      'Massachusetts',
    'Princeton':                    'Princeton',
    'Indiana Hoosiers':             'Indiana',
    'Ohio State Buckeyes':          'Ohio State',
    'Marshall Thundering':          'Marshall',
    'Marshall Thundering Herd':     'Marshall',
    'Central Arkansas':             'Central Arkansas',
    'UC Davis':                     'UC Davis',
    'Fresno State Bulldogs':        'Fresno State',
    'Northern Illinois':            'Northern Illinois',
    'UNLV Rebels':                  'UNLV',
    'Louisiana Tech Bulldogs':      'Louisiana Tech',
    'Rutgers Scarlet Knights':      'Rutgers',
    'Louisiana-Monroe':             'Louisiana-Monroe',
    'UAB Blazers':                  'UAB',
    'Western Michigan Broncos':     'Western Michigan',
    'Montana Grizzlies':            'Montana',
    'Eastern Kentucky':             'Eastern Kentucky',
    'Arkansas State Red Wolves':    'Arkansas State',
    'San Diego State Aztecs':       'San Diego State',
    'Utah Utes':                    'Utah',
    'North Texas Mean Green':       'North Texas',
    'Texas Southern':               'Texas Southern',
    'Louisville Cardinals':         'Louisville',
    'Akron Zips':                   'Akron',
    'Georgia Southern':             'Georgia Southern',
    'Western Carolina':             'Western Carolina',
    'Chattanooga':                  'Chattanooga',
    'Troy Trojans':                 'Troy',
    'Central Washington':           'Central Washington',
    'Air Force':                    'Air Force',
    'Mercer':                       'Mercer',
    'UTEP Miners':                  'UTEP',
    'Northern Arizona':             'Northern Arizona',
    'UTSA Roadrunners':             'UTSA',
    'USC Trojans':                  'USC',
    'Texas State Bobcats':          'Texas State',
    'Coastal Carolina':             'Coastal Carolina',
    'Harvard':                      'Harvard',
    'Central Michigan':             'Central Michigan',
    'Grand Valley State':           'Grand Valley State',
    'Houston Cougars':              'Houston',
    'Murray State':                 'Murray State',
    'Ball State Cardinals':         'Ball State',
    'Ball State':                   'Ball State',
    'Liberty Flames':               'Liberty',
    'Nicholls State Colonels':      'Nicholls State',
    'Ohio Bobcats':                 'Ohio',
    'Idaho State':                  'Idaho State',
    'San Jose State Spartans':      'San Jose State',
    'Rhode Island':                 'Rhode Island',
    'Oregon State Beavers':         'Oregon State',
    'Rice Owls':                    'Rice',
    'Eastern Michigan':             'Eastern Michigan',
    'Western Michigan':             'Western Michigan',
    'Simon Fraser':                 'Simon Fraser',
    'Navy':                         'Navy',
    'Army':                         'Army',
    'Eastern Washington':           'Eastern Washington',
    'Eastern Washington Eagles':    'Eastern Washington',
    'Western Oregon':               'Western Oregon',
    'Alabama State':                'Alabama State',
    'Kennesaw State Owls':          'Kennesaw State',
    'Texas-Permian Basin':          'Texas-Permian Basin',
    'Lamar':                        'Lamar',
    'Connecticut Huskies':          'Connecticut',
    'South Alabama Jaguars':        'South Alabama',
    'Montana State-Northern':       'Montana State-Northern',
    'Prairie View A&M':             'Prairie View A&M',
    'Tulsa Golden Hurricane':       'Tulsa',
    'Holy Cross':                   'Holy Cross',
    'Fordham':                      'Fordham',
    'Dartmouth':                    'Dartmouth',
    'Stephen F. Austin':            'Stephen F. Austin',
    'Portland State':               'Portland State',
    'Idaho':                        'Idaho',
    'Georgia Bulldogs':             'Georgia',
    'Georgetown':                   'Georgetown',
    'New Hampshire':                'New Hampshire',
    'Illinois State':               'Illinois State',
    'Austin Peay':                  'Austin Peay',
    'Yale':                         'Yale',
    'East Tennessee State':         'East Tennessee State',
    'Wyoming Cowboys':              'Wyoming',
}
 
# ── Set of known colleges (standardized names) ────────────────────────────────
# Anything NOT in this set after mapping will be flagged as non-college.
KNOWN_COLLEGES = set(SCHOOL_MAP.values())
 
# ── Apply standardization ─────────────────────────────────────────────────────
def standardize_school(raw_name):
    """Return (standardized_name, is_college) for a raw school string."""
    cleaned = raw_name.strip()
    standardized = SCHOOL_MAP.get(cleaned, cleaned)   # keep original if no mapping found
    is_college = standardized in KNOWN_COLLEGES
    return standardized, is_college
 
recruits_2019[['committed_school', 'Is_College']] = recruits_2019['committed_school'].apply(
     lambda x: pd.Series(standardize_school(x)))

In [22]:
recruits_2019[['name','committed_school', 'Is_College']]

,name,committed_school,Is_College
0,Spencer Rattler,Oklahoma,True
1,Jayden Daniels,Arizona State,True
2,Bo Nix,Auburn,True
3,Sam Howell,North Carolina,True
4,Max Duggan,TCU,True
...,...,...,...
242,Heath Williams,Northeast,False
243,Nick Lasher,Norco,False
244,Mitch McKenzie,Archbishop Moeller,False
245,Jack Baltz,Illinois State,True


Search 247 to see which are the actual colleges of players with colleges not in college list.

19 = N.C State
43 = Georgia
68 = USC
71 = NA
86 = NA
103 = NA
115 = NA
128 = NA
130 = NA
139 = NA
149 = NA
150 = NA
157 = NA
158 = NA
167 = NA
170 = NA
172 = NA
177 = NA
179 = NA
183 = NA
186 = NA
189 = NA
194 = NA
196 = NA
197 = NA
198 = NA
203 = NA
205 = NA
206 = NA
208 = NA
210 = NA
214 = NA
215 = NA
222 = NA
224 = NA
228 = NA
229 = NA
232 = NA
233 = NA
234 = NA
237 = NA
238 = NA
242 = NA
243 = NA
244 = NA


As seen above, looks like most of these are simply players that did not get a landing spot out of high school. The first few misses are likely due to the '.' in 'N.C. State' and some html formatting.